crates/ra_ide/src/completion/completion_item.rs

```diff
@@ -314,6 +346,54 @@ impl Completions {
     {
         items.into_iter().for_each(|item| self.add(item.into()))
     }
+
+    pub(crate) fn with_score_option(&mut self, score_option: ScoreOption) {
+        self.score_option = Some(score_option);
+    }
+
+    pub(crate) fn compute_score(&mut self, ctx: &CompletionContext) {
```

Comment:
Why do we need to delay the computation of scores until the end of completion process? A simpler approach seems to be to add a fully-computed score to `CompletionItem`.

Reply:
Yes I will move this part in completion_dot